Financial-Times/n-browsable-lists
# UG-990: Build component

### Description

This PR adds the logic to fetch the public list data and show it in the component.

- Adds the markup and logic to JSX component. When rendered server-side, it'll render empty and put the `concepts` prop into a DataEmbed component. And when rendered client-side, it'll expect a `listData` prop, and it'll render the list title and its articles.
- Adds the `main.js` file with the client-side logic. This logic will read the concepts list from the DataEmbed component, get the first concept and matches it against the`concept-list-map` to get a listId. It then fetches the data from `myFTClient` using the new endpoint `getPublicList`. Finally, it re-renders the server-side markup with the listData
- Adds the styling for the component to the `main.scss` file
- Updates the demo page to import the client-side logic through `main.js`.
- Configures the demo page to run on `https://local.ft.com:5005`. The demo page needs to run on ft.com to be able to use the session cookie and send it to MyFT Client
- Adds a script to generate local certs and start the page on `https://local.ft.com:5005`. 
- Adds a proxy to forward requests from `/__myft` to `https://ft.com/__myft` on the demo page. This change is needed because the demo page can't serve the MyFT Client requests to `__myft`, because it doesn't run on `next-router`.

### Ticket
[UG-990](https://financialtimes.atlassian.net/browse/UG-990)

### Screenshots
<img width="314" alt="n-browsable-lists" src="https://user-images.githubusercontent.com/7011304/195098528-dc5cfa64-e289-44aa-ae29-2ad0cd86c0e7.png">

### How do I test this code?

- Check out this branch locally 
- Run `npm run certs` to generate local certs
- Run the demo page with `npm run demo`
- Open the demo page on https://local.ft.com:5005/ and confirm that the component is showing the list title and its articles.

### Reminder
Have you completed these common tasks? (remove those that don't apply)

- [x] **Documentation** updated or created
- [x] **Design Review** ran past the designer 


### How we will review this PR
Our team uses [conventional comments](https://conventionalcomments.org/) to provide feedback.
